### ISSUE TITLE
Callable compared with is True not flagged

### ISSUE THREAD

**Issue by acampove** [microsoft/pyright#11442]
### Description

When a method or function is used in a conditional with `is True` (instead of being called), pyright does not report any warning or error. Since a callable can never be identical to the `True` singleton, this comparison is always `False`, making the branch dead code.

This is a type-level fact that pyright has all the information to detect: if the left-hand side of an `is True` expression is typed as a callable, the result is always `False`.

### Reproduction

```python
class Sample:
    def is_merged(self) -> bool:
        return True

sample = Sample()

if sample.is_merged is True:  # always False, but pyright reports no issue
    print("merged")
```

### Expected behavior

Pyright should report a warning/error such as:
> Comparison of a callable with `True` using `is` will always be `False`. Did you mean `sample.is_merged()`?

### Actual behavior

```
0 errors, 0 warnings, 0 informations
```

### Environment

- pyright version: 1.1.408
- Python version: 3.13

**Comment by rchiodo** [comment]
I don't believe this is something that Pyright would flag. This is more of a `ruff` type linting error. From a typing perspective your code is perfectly valid.
